libcore/ConnectServerDialog.vala

```diff
@@ -356,13 +358,19 @@ public class PF.ConnectServerDialog : Gtk.Dialog {
         info_bar.message_type = Gtk.MessageType.ERROR;
         info_label.label = error_message;
         show_info ();
+        show_connect_button ();
     }
 
     private void show_info () {
         show_connecting (false);
         info_bar.revealed = true;
     }
 
+    private void dismiss_info () {
+        info_label.label = "";
```

Comment:
Is this necessary? It seems like any time you show the infobar you'd be assigning a new message

Reply:
This was because of this line:
`        info_bar.revealed = !(valid || info_label.label.length < 1);`
which was put in the function `valid_entries ()`  synchronise the infobar with entry error state.  A side-effect of this was to sometimes show an empty info-bar e.g. at the start (the entries are invalid because there is no server entry but no error message has been produced.)